core/common/src/main/java/alluxio/heartbeat/HeartbeatTimer.java

```diff
@@ -11,13 +11,26 @@
 
 package alluxio.heartbeat;
 
+import alluxio.conf.PropertyKey;
 import alluxio.conf.Reconfigurable;
 
+import java.util.Map;
+
 /**
  * An interface for heartbeat timers. The {@link HeartbeatThread} calls the {@link #tick()} method.
  */
 public interface HeartbeatTimer extends Reconfigurable {
 
+  /**
+   * When this object needs to be reconfigured
+   * due to external configuration change etc.,
+   * this function will be invoked.
+   *
+   * @param changedProperties the changed properties
+   */
+  default void update(Map<PropertyKey, Object> changedProperties) {
+  }
```

Comment:
You are making this a default impl with empty body. This indicates that the `Reconfigurable` interface is *optional* to the `HeartbeatTimer` interface. How about not making `HeartbeatTimer` extend from `Reconfigurable` and instead making individual impls implement `Reconfigurable` if they need to ?